cadence/gateway/cadence/grpc_service.py

```diff
@@ -22,25 +29,36 @@ def __init__(self, host: str = 'localhost', port: int = 7833, timeout: int = 300
         self.timeout = timeout
         self.channel = grpc.insecure_channel(
             '{}:{}'.format(host, port))
-
         # stubs
         self.domain = service_domain_pb2_grpc.DomainAPIStub(self.channel)
         self.service_workflow = service_workflow_pb2_grpc.WorkflowAPIStub(self.channel)
 
+    def close(self):
+        self.channel.close()
+
+    def set_next_timeout_cb(self, cb: Callable):
+        raise NotImplementedError
+
     def list_domains(self, request: ListDomainsRequest) -> Tuple[ListDomainsResponse, object]:
         grpc_request = list_domains_request_dataclass_to_proto(request)
-        response = self.domain.ListDomains.with_call(
-            grpc_request,
-            metadata=self.metadata,
-            timeout=self.timeout
-        )
-        return (proto_list_domains_response_to_dataclass(response[0]), None) #TODO check how errors are process via tchannel
+        try:
+            response = self.domain.ListDomains.with_call(
+                grpc_request,
+                metadata=self.metadata,
+                timeout=self.timeout
+            )
+            return proto_list_domains_response_to_dataclass(response[0]), None
+        except grpc.RpcError as e:
+            return None, process_error(e)
 
     def start_workflow(self, request: StartWorkflowExecutionRequest) -> Tuple[StartWorkflowExecutionResponse, object]:
         grpc_request = start_workflow_execution_request_dataclass_to_proto(request)
-        response = self.service_workflow.StartWorkflowExecution.with_call(
-            grpc_request,
-            metadata=self.metadata,
-            timeout=self.timeout
-        )
-        return (start_workflow_execution_response_to_dataclass(response[0]), None) #TODO check how errors are process via tchannel
\ No newline at end of file
+        try:
+            response = self.service_workflow.StartWorkflowExecution.with_call(
+                grpc_request,
+                metadata=self.metadata,
+                timeout=self.timeout
+            )
+            return start_workflow_execution_response_to_dataclass(response[0]), None
+        except grpc.RpcError as e:
+            return None, process_error(e)
```

Comment:
how does it convert to error `WorkflowExecutionAlreadyStartedError` ?

Reply:
If starting same workflowID twice, you should see the error.

Reply:
<img width="1135" alt="Screen Shot 2021-12-23 at 12 12 32 PM" src="https://user-images.githubusercontent.com/2667294/147280499-7a9c6760-8ac9-4e93-85e1-b545cd35e996.png">

Reply:
Do you know where I can find all the status code so I can do the mapping for error?

Reply:
Forget my latest comment. I think I know how to do this mapping. Thanks!

Reply:
done 👍